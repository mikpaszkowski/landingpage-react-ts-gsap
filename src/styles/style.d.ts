import "styled-components";

interface IPalette {
  footerBackground: string;
  font: string;
  fontLight: string;
  playButton: string;
}

declare module "styled-components" {
  export interface DefaultTheme {
    palette: {
      common: {
        black: string;
        white: string;
        theme: string;
        background: string;
      };
      primary: IPalette;
      secondary: IPalette;
    };
  }
}
