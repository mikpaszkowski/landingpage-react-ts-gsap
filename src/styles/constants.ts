interface Fonts {
  primaryFont: string;
  secondaryFont: string;
}

interface BreakPoints {
  mobileSmall: string;
  mobileMedium: string;
  mobileLarge: string;
  tablet: string;
  tabletLarge: string;
  laptopSmall: string;
  laptopLarge: string;
}

export const fontTypes: Fonts = {
  primaryFont: "'Rozha One', serif",
  secondaryFont: "'Reem Kufi', sans-serif",
};

export const breakPoints: BreakPoints = {
  mobileSmall: "320px",
  mobileMedium: "375px",
  mobileLarge: "525px",
  tablet: "768px",
  tabletLarge: "1110px",
  laptopSmall: "1280px",
  laptopLarge: "1440px",
};

export {};
