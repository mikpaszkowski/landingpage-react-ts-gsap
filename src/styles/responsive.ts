import { breakPoints } from "./styleConstants";

interface DeviceResponsive {
  mobileMedium: string;
  laptopSmall: string;
  laptopLarge: string;
}

export const device: DeviceResponsive = {
  mobileMedium: `(max-width: ${breakPoints.mobileMedium})`,
  laptopSmall: `(max-width: ${breakPoints.laptopSmall})`,
  laptopLarge: `(max-width: ${breakPoints.laptopLarge})`,
};
