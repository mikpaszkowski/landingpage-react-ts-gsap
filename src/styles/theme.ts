import { DefaultTheme } from "styled-components";

export const lightTheme: DefaultTheme = {
  palette: {
    common: {
      black: "#131E07",
      white: "#FFFFFF",
      theme: "#FABD20",
      background: "#FEF4E6",
    },
    primary: {
      footerBackground: "#131E07",
      font: "#131E07",
      fontLight: "#E5E5E5",
      playButton: "#131E07",
    },
    secondary: {
      footerBackground: "#131E07",
      font: "#40462F",
      fontLight: "#7C8176",
      playButton: "#131E07",
    },
  },
};

export const darkTheme: DefaultTheme = {
  palette: {
    common: {
      black: "#131E07",
      white: "#FFFFFF",
      theme: "#FABD20",
      background: "#131E07",
    },
    primary: {
      footerBackground: "#131E07",
      font: "#FFFFFF",
      fontLight: "#E5E5E5",
      playButton: "#131E07",
    },
    secondary: {
      footerBackground: "#131E07",
      font: "#E5E5E5",
      fontLight: "#7C8176",
      playButton: "#131E07",
    },
  },
};
