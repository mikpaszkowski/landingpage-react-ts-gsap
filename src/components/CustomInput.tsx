import * as React from "react";
import styled from "styled-components";
import { colors, fontTypes } from "../styles/styleConstants";

const CustomInputWrapper = styled.input`
  position: relative;
  width: 35rem;
  height: 5rem;
  background-color: transparent;
  font-family: ${fontTypes.secondaryFont};
  color: ${colors.secondaryLight};
  font-style: normal;
  font-weight: normal;
  font-size: 1.8rem;
  line-height: 2.7rem;
  border: 1px solid ${colors.secondaryLight};
  border-radius: 8px;
  padding: 1.2rem 14rem 1.2rem 1.6rem;
`;

interface CustomInputProps {
  placeholder?: string;
}

export const CustomInput = ({ ...props }: CustomInputProps) => {
  return (
    <div>
      <CustomInputWrapper {...props} />
    </div>
  );
};
