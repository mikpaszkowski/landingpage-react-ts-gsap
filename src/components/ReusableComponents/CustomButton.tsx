import * as React from "react";
import styled from "styled-components";
import { fontTypes } from "../../styles/styleConstants";

const Button = styled.button`
  position: relative;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 0.7rem 3.1rem;
  font-family: ${fontTypes.secondaryFont};
  font-size: ${(props: ButtonProps) =>
    props.fontSize ? props.fontSize : "2.5rem"};
  font-style: normal;
  font-weight: normal;
  line-height: ${(props: ButtonProps) =>
    props.lineHeight ? props.lineHeight : "3.6rem"};
  color: ${({ theme }) => theme.palette.common.black};
  border: none;
  border-radius: ${(props: ButtonProps) =>
    props.borderRadius ? props.borderRadius : "8px"};
  background-color: ${(props: ButtonProps) =>
    props.color ? props.color : ({ theme }) => theme.palette.common.theme};
  cursor: pointer;
  transition: all 0.25s ease-in-out;
  overflow: hidden;
  will-change: transform;

  &::after {
    content: "";
    position: absolute;
    z-index: -1;
    bottom: -100%;
    left: 0;
    height: 100%;
    width: 100%;
    border-radius: ${(props: ButtonProps) =>
      props.borderRadius ? props.borderRadius : "8px"};
    background-color: black;
    color: ${({ theme }) => theme.palette.common.theme};
    transition: all 0.15s cubic-bezier(0.455, 0.03, 0.515, 0.955);
  }

  &:hover {
    box-shadow: 0px 1rem 2rem rgba(0, 0, 0, 0.25);
    color: ${({ theme }) => theme.palette.common.theme};

    &::after {
      transform: translateY(-100%);
    }
  }
`;

interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  color?: string;
  name?: string;
  lineHeight?: string;
  fontSize?: string;
  borderRadius?: string;
}

export const CustomButton = ({ name, ...props }: ButtonProps) => (
  <Button {...props}>{name}</Button>
);
