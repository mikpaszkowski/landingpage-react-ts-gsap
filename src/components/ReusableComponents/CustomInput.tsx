import * as React from "react";
import styled from "styled-components";
import { fontTypes } from "../../styles/styleConstants";
import { device } from "../../styles/responsive";

const CustomInputWrapper = styled.input`
  position: relative;
  width: 35rem;
  height: 5rem;
  background-color: transparent;
  font-family: ${fontTypes.secondaryFont};
  color: ${({ theme }) => theme.palette.secondary.fontLight};
  font-style: normal;
  font-weight: normal;
  font-size: 1.8rem;
  line-height: 2.7rem;
  border: 1px solid ${({ theme }) => theme.palette.secondary.fontLight};
  border-radius: 8px;
  padding: 1.2rem 14rem 1.2rem 1.6rem;

  @media ${device.mobileMedium} {
    width: 100%;
  }
`;

interface CustomInputProps {
  placeholder?: string;
}

export const CustomInput = ({ ...props }: CustomInputProps) => {
  return (
    <div>
      <CustomInputWrapper {...props} />
    </div>
  );
};
