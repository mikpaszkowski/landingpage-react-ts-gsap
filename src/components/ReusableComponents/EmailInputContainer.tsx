import * as React from "react";
import styled from "styled-components";
import { fontTypes } from "../../styles/styleConstants";
import { CustomInput } from "./CustomInput";
import { CustomButton } from "./CustomButton";
import { device } from "../../styles/responsive";

const EmailInputContainerWrapper = styled.div`
  position: relative;
  & > p {
    font-family: ${fontTypes.secondaryFont};
    font-style: normal;
    font-weight: normal;
    font-size: 1.8rem;
    line-height: 2.7rem;
    margin-bottom: 1.6rem;
    color: ${({ theme }) => theme.palette.secondary.fontLight};
  }
`;

const CustomButtonStyled = styled(CustomButton)`
  position: absolute;
  right: 0.8rem;
  bottom: 0.8rem;
  padding: 0.3rem 1.6rem 0.4rem 1.6rem;
  border-radius: 4px;

  @media ${device.mobileMedium} {
    position: relative;
    width: 100%;
    right: 0;
    bottom: 0;
    font-size: 2rem;
    padding: 1.6rem 8rem 1.6rem 8rem;
    margin: 1.6rem 0 6.3rem 0;
  }
`;

export const EmailInputContainer = () => {
  return (
    <EmailInputContainerWrapper>
      <p>Try Medli for free</p>
      <CustomInput placeholder="Enter your email" />
      <CustomButtonStyled
        fontSize="1.8rem"
        lineHeight="2.7rem"
        borderRadius="4px"
        name="Get started"
      />
    </EmailInputContainerWrapper>
  );
};
