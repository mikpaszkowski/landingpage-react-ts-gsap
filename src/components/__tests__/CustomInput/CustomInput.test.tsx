import { cleanup } from "@testing-library/react";
import "jest-styled-components";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import { CustomInput } from "../../ReusableComponents/CustomInput";
import { ThemeProvider } from "styled-components";
import { lightTheme } from "../../../styles/theme";

afterEach(() => {
  cleanup();
});

test("render correctly the <CustomInput /> component", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={lightTheme}>
        <CustomInput />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe("check CustomInput css properties", () => {
  test("should have placeholder displayed", () => {
    const props = {
      placeholder: "Enter your name...",
    };

    const wrapper = mount(
      <ThemeProvider theme={lightTheme}>
        <CustomInput {...props} />
      </ThemeProvider>
    ).find(CustomInput);
    expect(wrapper.at(0).props().placeholder).toEqual(props.placeholder);
  });

  test("should have border property", () => {
    const wrapper = mount(
      <ThemeProvider theme={lightTheme}>
        <CustomInput />
      </ThemeProvider>
    ).find(CustomInput);
    expect(wrapper).toHaveStyleRule("border", "1px solid #7C8176");
  });

  test("should have color property", () => {
    const wrapper = mount(
      <ThemeProvider theme={lightTheme}>
        <CustomInput />
      </ThemeProvider>
    ).find(CustomInput);
    expect(wrapper).toHaveStyleRule("color", "#7C8176");
  });
});
