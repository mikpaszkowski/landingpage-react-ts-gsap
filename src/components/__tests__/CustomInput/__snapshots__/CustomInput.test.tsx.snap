// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly the <CustomInput /> component 1`] = `
.c0 {
  position: relative;
  width: 35rem;
  height: 5rem;
  background-color: transparent;
  font-family: 'Reem Kufi',sans-serif;
  color: #7C8176;
  font-style: normal;
  font-weight: normal;
  font-size: 1.8rem;
  line-height: 2.7rem;
  border: 1px solid #7C8176;
  border-radius: 8px;
  padding: 1.2rem 14rem 1.2rem 1.6rem;
}

@media (max-width:375px) {
  .c0 {
    width: 100%;
  }
}

<div>
  <input
    className="c0"
  />
</div>
`;
